### DISCUSSION TITLE
Error - id input not resolving to model

### DISCUSSION THREAD

**Discussion by jhoskins98** [wire-elements/modal#103] [Q&A] [answered]
Using button with either single or double quotes, the resolution step does not change the id value into model instance.  

`<x-button wire:click="$emit('openModal', 'edit-driver', {{ json_encode(['driver' => $driver->id]) }})">Edit</x-button>`

Gives "Cannot assign int to property App\Http\Livewire\EditDriver::$driver of type App\Models\Driver (View: /home/vagrant/code/resources/views/vendor/livewire-ui-modal/modal.blade.php)" 

Fails on vendor/livewire/livewire/src/LifecycleManager.php:109.  How does the id get resolved to an actual model?

**Comment by jhoskins98** [comment, marked as answer]
Note on issue #60 say " change your property from:
public User $user; to public int|User $user; "

Seems like the Livewire hydration process does a straight assignment and later in the process does it get typed to the model.  

Not clear if the behavior is a Livewire issue or a modal-ui issue.

**Reply by PhiloNL** [reply]
This is how Livewire works I believe, so that's why it's needed to use int|User.

**Reply by dsbilling** [reply]
@PhiloNL Maybe this should be added to the documentation / readme?

**Reply by boranbar** [reply]
> @PhiloNL Maybe this should be added to the documentation / readme?

I agree. I had to same issue when i try to open an edit modal inside a livewire component.

This solution fixed my problem also.
